File: LB-Assignments/Assignment_35/program1.c
/*
Accept number from user and check whether
15th bit is on or off
*/
#include<stdio.h>

typedef unsigned int UNIT;
typedef int BOOL;

#define TRUE 1
#define FALSE 0

BOOL ChkBit(UNIT iNo)
{
    BOOL bRet = FALSE;

    UNIT iMask = 0x1;

    iMask = iMask << 14;

    if(iMask == (iNo & iMask))
    {
        bRet = TRUE;
    }
    return bRet;
}

int main()
{
    UNIT iValue = 0;

    printf("Enter Number : ");
    scanf("%d",&iValue);

    if(ChkBit(iValue))
    {
        printf("Bit is on\n");
    }
    else
    {
        printf("Bit is off\n");
    }

    return 0;
}

File: LB-Assignments/Assignment_37/program1.java
/*
Accept number from user and position also
check whether the bit at given position is on or off
*/
import java.util.*;

class Bitwise
{
    public int iNo, iPos;

    public Bitwise(int a, int b)
    {
        this.iNo = a;
        this.iPos = b;
    }

    public Boolean ChkBit()
    {
        boolean bRet = false;

        int iMask = 0x1;

        iMask = iMask << (iPos-1);

        if((iMask & iNo) == iMask)
        {
            bRet = true;
        }

        return bRet;
    }

}
class program1
{
    public static void main(String A[])
    {
        Scanner sobj = new Scanner(System.in);

        int iValue = 0, iPos = 0;
        boolean bRet = false;

        System.out.println("Enter number : ");
        iValue = sobj.nextInt();

        System.out.println("Enter bit position : ");
        iPos = sobj.nextInt();

        Bitwise bobj = new Bitwise(iValue,iPos);

        bRet = bobj.ChkBit();

        if(bRet)
        {
            System.out.println("Bit is on");
        }
        else
        {
            System.out.println("Bit is off");
        }

    }
}